README.md

```diff
@@ -18,25 +20,185 @@ $ npm install --save pipage
+
+// Find a stream in the pipeline:
+var index = pipeline.indexOf( stream )
+var lastIndex = pipeline.lastIndexOf( stream )
+```
+
+### Example
+
+```js
+var Pipage = require('pipage')
+var path = require('path')
+var zlib = require('zlib')
+var unbzip2 = require('unbzip2-stream')
+var xz = require('xz')
+
+var pipeline = new Pipage()
+
+switch( path.extname( filename ) ) {
+  case '.gz':
+    pipeline.prepend( zlib.createUnzip() );
+    break
+  case '.bz':
+  case '.bz2':
+    pipeline.prepend( unbzip2() );
+    break
+  case '.xz':
+    pipeline.prepend( new xz.Decompressor() )
+    break
 }
 
-new BlockReadStream( '/path/to/resin-os.img' )
+fs.createReadStream( filename )
   .pipe( pipeline )
-  .pipe( new BlockWriteStream( '/dev/rdisk2' ) )
-  .once( 'finish', function() {
-    // happy face
-  })
+  .pipe( fs.createWriteStream( destination ) )
+```
+
+### Events
+
+#### Error handling
+
+Errors from streams within the pipeline are listened to,
+and re-emitted on the pipeline itself, with an additional `.stream` property
+being set on the `error` object, which is the stream that emitted it:
+
+```js
+pipeline.on( 'error', function( error ) {
+  // This error originated from the pipeline's internal stream
+  // available as `error.stream`, not from the pipeline itself
+  if( error.stream ) {
+    // ...
+  } else {
+    // This error came from the pipeline itself
+  }
+})
+```
+
+#### Binding to contained stream's events
+
+```js
+var stream = pipeline.get(-1)
+
+// Re-emit a stream's events on the pipeline:
+pipeline.bind( stream, 'eventname' )
+pipeline.bind( stream, [ 'someevent', 'otherevent' ])
+
+// Remove re-emission of a stream's event:
+pipeline.unbind( stream, 'eventname' )
+
+// Stop re-emission of all of stream's events on the pipeline:
+pipeline.unbindAll( stream )
+```
+
+##### Example
+
+```js
+var Pipage = require('pipage')
+
+// Let's say we have a stream which emits an event
+// we want to capture without having to get a reference to
+// that particular stream:
+module.exports = function createPipeline() {
+
+  var checksumStream = createChecksumStream( 'sha256', 'md5' )
+  var pipeline = new Pipage([ checksumStream ])
+
+  // This will cause the pipeline to re-emit
+  // the 'checksums' event from the `checksumStream`
+  pipeline.bind( checksumStream, 'checksums' )
+
+  // Add some more fancy things to the pipeline...
+
+  return pipeline
+
+}
+```
+
+```js
+var createPipeline = require('./create-pipeline')
+var pipeline = createPipeline()
+
+// Now we can listen for the bound event directly on the pipeline
+pipeline.on( 'checksums', function( checksums ) {
+  // Validate the checksums, etc...
+})
+
+fs.createReadStream( filename )
+  .pipe( pipeline )
+  .resume()
+```
+
+### Nested Pipelines
+
+Since pipelines are duplex streams, and contain duplex streams, they can be nested arbitrarily:
+
+```js
+var pipeline = new Pipage([
+  new Pipage(),
+  new Pipage([
+    new Pipage()
+  ])
+])
```

Comment:
I guess in this (obviously artificial) example, you'd only be able to get each Pipeline's streams by index, rather than by reference? And I guess even in this nested structure, data would still flow through the pipeline in a linear order?

Hmm, I wonder if something like this should/could be better supported by a method that did something like:
```js
appendPipeline(other) {
    this.splice(this._streams.length, 0, other.splice(0, other.length))
}
```
I've got no idea what kind of mangling would be needed to transfer all the custom-bound events over though... (or even if this crazy suggestion actually makes sense?!?)

P.S. Should the `length` property be added to the API docs? Should there be some public method of enumerating the custom events bound to a pipeline?